### PR TITLE
smbios: support UNIX-like operating systems, README intro

### PR DESCRIPTION
**Pull request by mdlayher**
Adds support for all UNIX-like systems that expose `/dev/mem`, and also beefs up the README.

### CODE REVIEW

File: smbios/stream_memory.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+		return nil, nil, err
+	}
+
+	// Seek to the start of the SMBIOS table.
+	tableAddr, tableSize := ep.Table()
+	if _, err := rs.Seek(int64(tableAddr), io.SeekStart); err != nil {
+		return nil, nil, err
+	}
+
+	// Make a copy of the memory so we don't return a handle to system memeory
```
Comment (digitalocean-engineering) [inline smbios/stream_memory.go]: nit: memory

**Comment by digitalocean-engineering** [comment]
lgtm

**Comment by mdlayher** [comment]
Will fix nit also get a review from a teammate, and then merge.

**Pull request merged**